js/main.js

```diff
@@ -20,7 +27,86 @@ function handleSubmit(event) {
   data.entries.unshift(entry);
   $photoUrl.setAttribute('src', 'images/placeholder-image-square.jpg');
   $entryForm.reset();
+  var newEntry = generateEntryDom(entry);
+  $ul.prepend(newEntry);
+  $noEntriesText.className = 'hidden';
+  swapView(data.view);
+}
+
+// creates DOM tree for each new entry
+
+function generateEntryDom(entry) {
+  var entriesLi = document.createElement('li');
+  entriesLi.setAttribute('class', 'entries-li');
+
+  var entriesRowDiv = document.createElement('div');
+  entriesRowDiv.setAttribute('class', 'row');
+  entriesLi.appendChild(entriesRowDiv);
+
+  var imageColumnHalf = document.createElement('div');
+  imageColumnHalf.setAttribute('class', 'column-half');
+  entriesRowDiv.appendChild(imageColumnHalf);
+
+  var entriesImage = document.createElement('img');
+  entriesImage.setAttribute('class', 'column-full img-box');
+  entriesImage.setAttribute('src', entry.photoUrl);
+  imageColumnHalf.appendChild(entriesImage);
+
+  var entriesColumnHalf = document.createElement('div');
+  entriesColumnHalf.setAttribute('class', 'column-half');
+  entriesRowDiv.appendChild(entriesColumnHalf);
+
+  var entriesName = document.createElement('h2');
+  entriesName.setAttribute('class', 'entries-name');
+  entriesName.textContent = entry.title;
+  entriesColumnHalf.appendChild(entriesName);
+
+  var entriesNotes = document.createElement('p');
+  entriesNotes.setAttribute('class', 'entries-notes');
+  entriesNotes.textContent = entry.notes;
+  entriesColumnHalf.appendChild(entriesNotes);
+
+  return entriesLi;
+}
+
+// appends new entry DOM tree to the UL
+
+function appendDom(entry) {
+  swapView(data.view);
+  if (data.entries.length !== 0) {
+    $noEntriesText.className = 'hidden';
+  }
+  for (var i = 0; i < data.entries.length; i++) {
+    var returnObject = generateEntryDom(data.entries[i]);
+    $ul.appendChild(returnObject);
+  }
+}
+
+// view swapping functions
+
+function swapView(string) {
+  data.view = string;
+  for (var i = 0; i < $views.length; i++) {
+    if ($views[i].getAttribute('data-view') === string) {
+      $views[i].className = 'view';
+    } else {
+      $views[i].className = 'hidden';
+    }
+  }
+}
+
+function dataView(event) {
+  var dataViewValue = event.target.getAttribute('data-view');
+  if (dataViewValue === null) {
+    return;
+  }
+  swapView(dataViewValue);
+  if (!data.view) {
+    swapView(dataViewValue);
+  }
```

Comment:
This shouldn't be necessary, you are doing the work in the conditional's codeblock on the line above, this would always be a repetition of an action that has already been done.